web/src/lib/components/admin-page/settings/admin-settings.svelte

```diff
@@ -0,0 +1,80 @@
+<svelte:options accessors />
+
+<script lang="ts">
+  import { SystemConfigDto, api } from '@api';
+  import {
+    notificationController,
+    NotificationType,
+  } from '$lib/components/shared-components/notification/notification';
+  import { handleError } from '$lib/utils/handle-error';
+  import _ from 'lodash';
+  import type { SettingsEventType } from './admin-settings';
+  import { createEventDispatcher, onMount } from 'svelte';
+
+  export let config: SystemConfigDto;
+
+  let savedConfig: SystemConfigDto;
+  let defaultConfig: SystemConfigDto;
+
+  const dispatch = createEventDispatcher<{ save: void }>();
+
+  const handleReset = async (detail: SettingsEventType['reset']) => {
+    if (detail.default) {
+      await resetToDefault(detail.configKeys);
+    } else {
+      await reset(detail.configKeys);
+    }
+  };
+
+  const handleSave = async (config: Partial<SystemConfigDto>) => {
+    try {
+      const result = await api.systemConfigApi.updateConfig({
+        systemConfigDto: {
+          ...savedConfig,
+          ...config,
+        },
+      });
+
+      savedConfig = { ...result.data };
+      notificationController.show({ message: 'Settings saved', type: NotificationType.Info });
+
+      dispatch('save');
+    } catch (error) {
+      handleError(error, 'Unable to save settings');
+    }
+  };
+
+  const reset = async (configKeys: Array<keyof SystemConfigDto>) => {
+    const { data: resetConfig } = await api.systemConfigApi.getConfig();
+
+    config = { ...config, ..._.get(resetConfig, configKeys) };
+
+    notificationController.show({
+      message: 'Reset settings to the recent saved settings',
+      type: NotificationType.Info,
+    });
+  };
+
+  const resetToDefault = async (configKeys: Array<keyof SystemConfigDto>) => {
+    notificationController.show({
+      message: 'Reset settings to default',
+      type: NotificationType.Info,
+    });
+
+    config = { ...config, ..._.get(defaultConfig, configKeys) };
+    savedConfig = { ...savedConfig, ..._.get(defaultConfig, configKeys) };
```

Comment:
Same here, it seems like resetting to the entire config object might work just as well.